### ISSUE TITLE
performance

### ISSUE THREAD

**Issue by sbabashahi** [jobguywork/backend#18]
Improve performance of /public/home/

**Comment by MAM-SYS** [comment]
hello friends
i was reviewing the HomeView  code that i faced something
it is not a big problem but i decided to share it with you
https://github.com/jobguywork/backend/blob/e88b6e6173a00524a80f35ddcb3537f4de7bd89f/company/views.py#L850
i faced the ***pagination_permission*** function that is getting index argument and just return it again with no changing
as i said it is not a big problem at all but this function is repeating many times in the code base
so i just wanted to mention that
thanks again for this amazing project

**Comment by sbabashahi** [comment]
thanks for your cooperation :)
it may be a confusing name but if you check the pagination_permission it check for staff and non staff users to limit size of pagination for non staff users by allowed_size for now. so basically it doing some job :))
but as getting index! i don't sure about it but it may be just like a reserved place to have a hook checking index for future requirements.
as you said it's not a big deal :)

**Comment by MAM-SYS** [comment]
thanks for answering
yes, of course
this function is doing right, i was just noticing about index argument in this function that is just passing in and returning back with no change in any condition
but excuse me for misunderstanding if i am wrong

**Comment by sbabashahi** [comment]
You are not wrong, think about it as a bug  that have no harm now but in future have potential to be a feature :))
